mods/d2k/maps/harkonnen-07/harkonnen07.lua

```diff
@@ -0,0 +1,279 @@
+	end)
+
+	if #targets > 0 then
+		AHiTechFactory.SendAirstrike(Utils.Random(targets).CenterPosition, true, 0)
+	end
+
+	Trigger.AfterDelay(DateTime.Minutes(5), function()
+		SendAirStrike()
+	end)
+end
+
+Tick = function()
+	if player.HasNoRequiredUnits() then
+		atreides_main.MarkCompletedObjective(KillHarkonnen1)
+		atreides_small.MarkCompletedObjective(KillHarkonnen2)
+		corrino.MarkCompletedObjective(KillHarkonnen3)
+	end
+
+	if atreides_main.HasNoRequiredUnits() and atreides_small.HasNoRequiredUnits() and not player.IsObjectiveCompleted(KillAtreides) then
+		Media.DisplayMessage("The Atreides have been annihilated!", "Mentat")
+		player.MarkCompletedObjective(KillAtreides)
+	end
+
+	if corrino.HasNoRequiredUnits() and not player.IsObjectiveCompleted(KillCorrino) then
+		Media.DisplayMessage("The Emperor have been annihilated!", "Mentat")
+		player.MarkCompletedObjective(KillCorrino)
+	end
+
+	if AConYard2.Owner == player and not player.IsObjectiveCompleted(CaptureAtreidesConYard) then
+		player.MarkCompletedObjective(CaptureAtreidesConYard)
+	end
+
+	if (HEngineer.IsDead or AConYard2.IsDead) and not player.IsObjectiveCompleted(CaptureAtreidesConYard) then
+		player.MarkFailedObjective(CaptureAtreidesConYard)
+	end
+
+	if (AHiTechFactory.IsDead or AHiTechFactory.Owner ~= atreides_main) and not HiTechIsDead then
+		Media.DisplayMessage("High Tech Factory neutralized! Atreides cut off from Imperial reinforcement!", "Mentat")
+		HiTechIsDead = true
+	end
+
+	if DateTime.GameTime % DateTime.Seconds(10) == 0 and LastHarvesterEaten[ordos_main] then
+		local units = ordos_main.GetActorsByType("harvester")
+
+		if #units > 0 then
+			LastHarvesterEaten[ordos_main] = false
+			ProtectHarvester(units[1], ordos_main, AttackGroupSize[Difficulty])
+		end
+	end
+
+	if DateTime.GameTime % DateTime.Seconds(10) == 0 and LastHarvesterEaten[ordos_small] then
+		local units = ordos_small.GetActorsByType("harvester")
+
+		if #units > 0 then
+			LastHarvesterEaten[ordos_small] = false
+			ProtectHarvester(units[1], ordos_small, AttackGroupSize[Difficulty])
+		end
+	end
+end
+
+WorldLoaded = function()
+	atreides_main = Player.GetPlayer("Atreides Main Base")
+	atreides_small = Player.GetPlayer("Atreides Small Base")
+	corrino = Player.GetPlayer("Corrino")
+	player = Player.GetPlayer("Harkonnen")
+
+	InitObjectives(player)
+	CaptureAtreidesConYard = player.AddPrimaryObjective("Capture the Atreides Construction Yard.")
+	KillAtreides = player.AddPrimaryObjective("Destroy the Atreides.")
+	KillCorrino = player.AddPrimaryObjective("Destroy the Corrino.")
+	KillHarkonnen1 = atreides_main.AddPrimaryObjective("Kill all Harkonnen units.")
+	KillHarkonnen2 = atreides_small.AddPrimaryObjective("Kill all Harkonnen units.")
+	KillHarkonnen3 = corrino.AddPrimaryObjective("Kill all Harkonnen units.")
+
+	Camera.Position = HEngineer.CenterPosition
+	AtreidesAttackLocation = AConYard2.Location
+
+	Trigger.AfterDelay(DateTime.Minutes(5), function()
+		SendAirStrike()
```

Comment:
This can be simplified to `Trigger.AfterDelay(DateTime.Minutes(5), SendAirStrike)`.